pwndbg/dbg/gdb.py

```diff
@@ -17,6 +17,61 @@
 from pwndbg.gdblib import load_gdblib
 
 
+class GDBFrame(pwndbg.dbg_mod.Frame):
+    def __init__(self, inner: gdb.Frame):
+        self.inner = inner
+
+    @override
+    def evaluate_expression(self, expression: str) -> pwndbg.dbg_mod.Value:
+        selected = gdb.selected_frame()
+        restore = False
+        if selected != self.inner:
+            self.inner.select()
+            restore = True
```

Comment:
It would be neat if we had a Python ContextManager (https://realpython.com/python-with-statement/) for switching to a different frame (but this is out of scope for this PR).

Reply:
I agree! Actually, this might be a great idea to incorporate into the next PR, which among other things has that function wrapping GDB errors in an error type of our own. Using `with` should make exiting our temporarily selected frame - regardless of success or error - pretty clean and concise.